### ISSUE TITLE
gulp-notify and gulp-plumber, ReferenceError: through is not defined

### ISSUE THREAD

**Issue by richardegil** [mikaelbr/gulp-notify#86]
In the https://www.npmjs.com/package/gulp-notify#notes-tip section you provide an example of using gulp-notify and gulp-plumber together. Every time I try to run it I get the following error: 

ReferenceError: through is not defined

I have tried using gulp-through and through-pipes to help resolve this but it just leads to more errors. Am I missing something? Am I using the wrong through library?

Here is my gulpfile
https://gist.github.com/richardegil/9c513f8819e8ef25480c

**Comment by mikaelbr** [comment]
Hi! Try installing through2 and require it, see if that helps.

**Comment by richardegil** [comment]
Thanks for the quick response!

Now I am getting:

gulp-notify: [Error running Gulp] Error: Invalid non-string/buffer chunk

**Comment by mikaelbr** [comment]
Thaat might be due to your chunks being objects or other non strings. Try usingt through.obj instead to use through in object mode. The example might be broken, Im afraid

**Comment by richardegil** [comment]
Yeah, it seems to be broken. I will try to take some time to figure out a workaround but will have to abandon this for the time being.

Thanks for the super quick responses!

**Comment by ElliottRoche** [comment]
I'm getting the same Reference error and also the same ```gulp-notify: [Error running Gulp] Error: Invalid non-string/buffer chunk.``` Anyone have a fix at this point?

**Comment by ElliottRoche** [comment]
For what it's worth, I got what I needed by using:

```.pipe(plumber({errorHandler: notify.onError("Error: <%= error %> <%= error.message %>")}))```

No through2 or anything like that.

**Comment by mikaelbr** [comment]
No, that is the intended use for normal error handling. See [onError docs](https://github.com/mikaelbr/gulp-notify/#notifyonerror) or [tips on use with plumber](https://github.com/mikaelbr/gulp-notify/#notestip). The through use in these examples are only to fill a pipe with errors to illustrate how it works.

**Comment by Zundrium** [comment]
Maybe that's what @ElliottRoche wants, just a normal error for syntax problems or anything like that. I'm experiencing the same issue, then installed the through2 package. Also had the `Invalid non-string/buffer chunk` error. And solved it with plumber :)

**Comment by gurmukhp** [comment]
Could you share some code showing it solved with through2 and plumber? I get the non-string buffer check error every time.
